addons/point_of_sale/static/src/app/screens/receipt_screen/receipt_screen.js

```diff
@@ -94,15 +107,14 @@ export class ReceiptScreen extends Component {
             });
             return Promise.reject();
         }
-        const ticketImage = await this.renderer.toJpeg(
-            OrderReceipt,
-            {
-                data: this.pos.orderExportForPrinting(this.pos.get_order()),
-                formatCurrency: this.env.utils.formatCurrency,
-            },
-            { addClass: "pos-receipt-print p-3" }
-        );
-        await this.pos.data.call("pos.order", action, [[order.id], this.state.input, ticketImage]);
+        const fullTicketImage = await this.generateTicketImage();
```

Comment:
can you make `generateTicketImage` be a lambda function here, instead of declaring it in the class ?